kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/Coordinator.java

```diff
@@ -255,6 +288,77 @@ private void commitToTable(
     }
   }
 
+  private static String dataFilePath(DataFile dataFile) {
+    return dataFile.path().toString();
+  }
+
+  private static String deleteFilePath(DeleteFile deleteFile) {
+    return deleteFile.path().toString();
+  }
+
+  private <T> List<T> deduplicateBatch(
+      List<T> files,
+      Function<T, String> getPathFn,
+      String fileType,
+      UUID commitId,
+      TableIdentifier tableIdentifier) {
+    return deduplicate(
+        files,
+        getPathFn,
+        (numDuplicatedFiles, path) ->
+            String.format(
+                "Detected %d %s files with the same path=%s across payloads during commitId=%s for table=%s",
+                numDuplicatedFiles,
+                fileType,
+                path,
+                commitId.toString(),
+                tableIdentifier.toString()));
+  }
+
+  private <T> List<T> deduplicatePayload(
+      List<T> files,
+      Function<T, String> getPathFn,
+      String fileType,
+      UUID commitId,
+      TableIdentifier tableIdentifier) {
+    return deduplicate(
+        files,
+        getPathFn,
+        (numDuplicatedFiles, path) ->
+            String.format(
+                "Detected %d %s files with the same path=%s in payload with commitId=%s for table=%s",
```

Comment:
If we see this log message, it means the workers have somehow included the same file twice in a single CommitResponse.